internal/sig/trap.go

```diff
@@ -7,14 +7,14 @@ import (
 )
 
 // Trap cancels the context based on a signal
-func Trap(ctx context.Context, signals ...os.Signal) (context.Context, context.CancelFunc) {
+func Trap(ctx context.Context, signals ...os.Signal) context.Context {
 	ret, cancel := context.WithCancel(ctx)
 	ch := make(chan os.Signal, len(signals))
 	go func() {
 		<-ch
-		cancel()
 		signal.Stop(ch)
+		cancel()
 	}()
 	signal.Notify(ch, signals...)
-	return ret, cancel
+	return ret
```

Comment:
This was done because if a command finished (got to the end of the `Run(ctx) error` function), its context would be canceled. 

This was surprising behavior since this package was primarily just meant to cancel upon a signal. To reduce confusion, this PR removes the ability to cancel a trap manually. 

You could still wrap the trap ctx in a cancellable context if you want the old behavior